bitrepository-integrity-service/src/main/java/org/bitrepository/integrityservice/web/RestIntegrityService.java

```diff
@@ -83,146 +83,209 @@ public RestIntegrityService() {
+     * REST endpoint to get the list of present files on a pillar in a given collection.
      *
-     * @param collectionID, the collectionID from which to return checksum errors
-     * @param pillarID,     the ID of the pillar in the collection from which to return checksum errors
-     * @param pageNumber,   the page number for calculating offsets (@see pageSize)
-     * @param pageSize,     the number of checksum errors per page.
+     * @param collectionID The collection ID from which to return present file list.
+     * @param pillarID     The ID of the pillar in the collection from which to return present file list
+     * @return TODO: Missing
      */
     @GET
-    @Path("/getChecksumErrorFileIDs/")
+    @Path("/getAllFileIDs/")
     @Produces(MediaType.APPLICATION_JSON)
-    public StreamingOutput getChecksumErrors(
+    public HashMap<String, List<String>> getAllFileIDs(
             @QueryParam("collectionID")
                     String collectionID,
             @QueryParam("pillarID")
-                    String pillarID,
-            @QueryParam("pageNumber")
-                    int pageNumber,
-            @DefaultValue("100")
-            @QueryParam("pageSize")
-                    int pageSize) {
+                    String pillarID) {
+
+        IntegrityIssueIterator it = model.getFilesOnPillar(pillarID, 0, Integer.MAX_VALUE, collectionID);
 
-        int firstID = (pageNumber - 1) * pageSize;
+        List<String> fileList;
+        if (it != null) {
+            fileList = List.of(JSONStreamingTools.StreamIntegrityIssues(it).toString());
+        } else {
+            throw new WebApplicationException(
+                    Response.status(Response.Status.NO_CONTENT).entity("Failed to get missing files from database")
+                            .type(MediaType.TEXT_PLAIN).build());
+        }
 
-        return streamPartFromLatestReport(ReportPart.CHECKSUM_ISSUE, collectionID, pillarID, firstID, pageSize);
+        return new HashMap<>() {{
+            put(pillarID, fileList);
+        }};
     }
 
     /**
-     * Method to get the list of missing files per pillar in a given collection.
+     * REST endpoint to get the list of missing files for a pillar in a given collection.
      *
-     * @param collectionID, the collectionID from which to return missing files
-     * @param pillarID,     the ID of the pillar in the collection from which to return missing files
-     * @param pageNumber,   the page number for calculating offsets (@see pageSize)
-     * @param pageSize,     the number of checksum errors per page.
+     * @param collectionID The collection ID from which to return missing files.
+     * @return Returns a {@link HashMap} mapping the given pillar to its missing files.
      */
     @GET
     @Path("/getMissingFileIDs/")
     @Produces(MediaType.APPLICATION_JSON)
-    public StreamingOutput getMissingFileIDs(
+    public HashMap<String, List<String>> getMissingFileIDs(
```

Comment:
The methods `getMissingFileIDs` and `getAllMissingFilesInformation` are the same. Do we need both? If so, could they share their implementation? (Intellij IDEA drew my attention to this one.) There could be opportunities fr even more code reuse in the file.